sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -132,12 +131,11 @@ case class DataSource(
               }
           }
         } catch {
-          case e: NoClassDefFoundError => // This one won't be caught by Scala NonFatal
-            // NoClassDefFoundError's class name uses "/" rather than "." for packages
-            val className = e.getMessage.replaceAll("/", ".")
-            if (spark2RemovedClasses.contains(className)) {
-              throw new ClassNotFoundException(s"$className was removed in Spark 2.0. " +
-                "Please check if your library is compatible with Spark 2.0", e)
+          case e: NoClassDefFoundError =>
+            // e.getMessage is the class name of provider2. Instead, we use provider here.
+            if (spark2RemovedClasses.contains(provider)) {
```

Comment:
You should not change this. If `provider` is not found, `loadClass` will throw `ClassNotFoundException`. If a class used by provider are not found, `NoClassDefFoundError` will be thrown.